core/consensus/grandpa/vote_tracker.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_CONSENSUS_GRANDPA_VOTE_TRACKER_HPP
+#define KAGOME_CORE_CONSENSUS_GRANDPA_VOTE_TRACKER_HPP
+
+#include <unordered_map>
+
+#include "consensus/grandpa/structs.hpp"
+
+namespace kagome::consensus::grandpa {
+
+  template <typename Vote>
+  struct VoteTracker {
+    enum class AddVoteResult { SUCCESS, DUPLICATED, VOTED_TWICE };
+
+    /**
+     * @brief Adds a vote to a storage.
+     * @param id peer id
+     * @param n vote
+     * @return SUCCESS if peer voted once, DUPLICATED if we received duplicated
+     * vote, VOTED_TWICE if peer created two different votes
+     */
+    AddVoteResult addVote(Id id, const Vote &n) {
```

Comment:
implementatin should be moved to *impl.cpp file

Reply:
it is template